Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -335,74 +456,102 @@ struct UpdateMany: TestOperation {
+            session: sessions[self.session ?? ""]
         )
         return TestOperationResult(from: result)
     }
 }
 
 struct DeleteMany: TestOperation {
+    let session: String?
     let filter: Document
     let options: DeleteOptions
 
-    private enum CodingKeys: String, CodingKey { case filter }
+    private enum CodingKeys: String, CodingKey { case session, filter }
 
     init(from decoder: Decoder) throws {
         self.options = try DeleteOptions(from: decoder)
         let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = try container.decodeIfPresent(String.self, forKey: .session)
         self.filter = try container.decode(Document.self, forKey: .filter)
     }
 
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessions: [String: ClientSession]) throws -> TestOperationResult? {
         guard case let .collection(collection) = target else {
             throw TestError(message: "collection not provided to deleteMany")
         }
-        let result = try collection.deleteMany(self.filter, options: self.options, session: session)
+        let result =
+            try collection.deleteMany(self.filter, options: self.options, session: sessions[self.session ?? ""])
         return TestOperationResult(from: result)
     }
 }
 
 struct DeleteOne: TestOperation {
+    let session: String?
     let filter: Document
     let options: DeleteOptions
 
-    private enum CodingKeys: String, CodingKey { case filter }
+    private enum CodingKeys: String, CodingKey { case session, filter }
 
     init(from decoder: Decoder) throws {
         self.options = try DeleteOptions(from: decoder)
         let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = try container.decodeIfPresent(String.self, forKey: .session)
         self.filter = try container.decode(Document.self, forKey: .filter)
     }
 
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessions: [String: ClientSession]) throws -> TestOperationResult? {
         guard case let .collection(collection) = target else {
             throw TestError(message: "collection not provided to deleteOne")
         }
-        let result = try collection.deleteOne(self.filter, options: self.options, session: session)
+        let result = try collection.deleteOne(self.filter, options: self.options, session: sessions[self.session ?? ""])
         return TestOperationResult(from: result)
     }
 }
 
 struct InsertOne: TestOperation {
+    let session: String?
     let document: Document
 
-    func execute(on target: TestOperationTarget, session _: ClientSession?) throws -> TestOperationResult? {
+    private enum CodingKeys: String, CodingKey { case session, document }
+
+    init(from decoder: Decoder) throws {
```

Comment:
I think this custom initializer isn't required. After looking over this file again, it seems its necessary for operations that decode top level arguments into an optoins struct (e.g. `DeleteOne` does this).

Reply:
Done.